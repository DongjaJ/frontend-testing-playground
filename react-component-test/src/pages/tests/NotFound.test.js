import { withRouter } from '../../tests/utils';
import { render, screen } from '@testing-library/react';
import { Route } from 'react-router-dom';
import NotFound from '../NotFound';
import renderer from 'react-test-renderer';

describe('Not Found page', () => {
  it('render correctly', () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<NotFound />} />)
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('render with Not Found text', () => {
    render(withRouter(<Route path="/" element={<NotFound />} />));
    expect(screen.getByText('Not Found!')).toBeInTheDocument();
  });
});
