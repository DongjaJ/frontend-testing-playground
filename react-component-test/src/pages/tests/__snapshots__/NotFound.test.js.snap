// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found page render correctly 1`] = `
<div>
  Not Found!
</div>
`;
